cpp/include/raft/neighbors/ball_cover-inl.cuh

```diff
@@ -295,6 +294,103 @@ void knn_query(raft::resources const& handle,
   }
 }
 
+/**
+ * @brief Computes epsilon neighborhood for the L2 distance metric using rbc
+ *
+ * @tparam value_t   IO and math type
+ * @tparam idx_t    Index type
+ *
+ * @param[in] handle raft handle for resource management
+ * @param[in] index ball cover index which has been built
+ * @param[out] adj    adjacency matrix [row-major] [on device] [dim = m x n]
+ * @param[out] vd     vertex degree array [on device] [len = m + 1]
+ *                    `vd + m` stores the total number of edges in the adjacency
+ *                    matrix. Pass a nullptr if you don't need this info.
+ * @param[in]  x      first matrix [row-major] [on device] [dim = m x k]
+ * @param[in]  m      number of rows in x
+ * @param[in]  n      number of columns in x and index
+ * @param[in]  eps    defines epsilon neighborhood radius
+ */
+template <typename idx_t, typename value_t, typename int_t, typename matrix_idx = std::uint32_t>
+void epsUnexpL2NeighborhoodRbc(raft::resources const& handle,
+                               const BallCoverIndex<idx_t, value_t, int_t, matrix_idx>& index,
```

Comment:
Same as above wrt pointr-based APIs. Please use mdspan for any APIs that users will be interacting with. Anything in `detail` is fine staying pointer-based for now.